### PR TITLE
chore: remove `is_monorepo` and `path_to_yaml` from `GenerationConfig`

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Ronvert `is_monorepo` to a method in `GenerationConfig`.
- Remove `path_to_yaml` from `GenerationConfig`.

These two parameters are not used for library generation. Therefore, they should not be a parameter in `GenerationConfig`.

### CODE REVIEW

File: library_generation/cli/entry_point.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def generate(
         # baseline_generation_config is not specified.
         # Do not generate pull request description.
         generate_from_yaml(
+            config_path=current_generation_config,
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:116]: Not part of this PR, `current_generation_config` and `baseline_generation_config` are actually paths not the content, so probably better to rename them as something like `current_generation_config_path`.

**Review by blakeli0 — APPROVED** [review]

**Review by diegomarquezp — APPROVED** [review]

File: library_generation/cli/entry_point.py (hunk 1 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py:116]: Raised #2638.
